### PR TITLE
upgraded strimzi to 0.41.0 and improved strimzi installation to be able to select the version for system tests

### LINKED ISSUE #1209: [System Tests] Fix strimzi installation to select the proper version and not installing the latest one
**Describe the bug**
After the changes included in Strimzi 0.41.0 not supporting `KafkaNodePools` feature gate, our tests cases are broken because we always install the latest version.

**Expected behavior**
We should install the currently supported version taken from the main `pom.xml` file and modify the installation from the url to use helm charts/github releases selecting the proper version.

### PR DESCRIPTION
**Pull request by franvila**
### Type of change

- Bugfix

### Description

Fixes #1209. Replaced the strimzi installation by using helm charts to be able to select the proper strimzi version and not using the latest always, avoiding new breaks in the tests.

### Additional Context

After the release of Strimzi 0.41.0 that does not accept feature gates like `KafkaNodePools`, the strimzi installation was broken for all tests, so we need to assure repeatability and control the version upgrade.

### Checklist

- [ ] Write tests
- [X] Make sure all tests pass
- [ ] Review performance test results. Ensure that any degradations to performance numbers are understood and justified.
- [x] Make sure all Sonarcloud warnings are addressed or are justifiably ignored.
- [X] Update documentation
- [x] Reference relevant issue(s) and close them after merging
- [ ] For user facing changes, update CHANGELOG.md (remember to include changes affecting the API of the test artefacts too).

### CODE REVIEW

**Comment by franvila** [comment]
@strimzi-ci run system tests

File: kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java [hunk 1]
```diff
@@ -26,21 +33,24 @@ public static FileAttribute<Set<PosixFilePermission>> getDefaultPosixFilePermiss
...
-    public static String concatStringIfValueDontExist(String originalString, String stringToConcat, String stringSeparator) {
-        if (originalString.isEmpty() || originalString.isBlank()) {
-            stringSeparator = "";
+    @NonNull
+    public static Path getResourcesPath(String fileName) {
+        Path overrideFile;
+        var resource = TestUtils.class.getClassLoader().getResource(fileName);
+        try {
+            if (resource == null) {
+                throw new IllegalStateException("Cannot find resource " + fileName + " on classpath");
```
Comment (k-wall) [inline kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java:47]: It is not really an IllegalStateException.  IllegalArgumentException might be better.

File: kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java [hunk 2]
```diff
@@ -26,21 +33,24 @@ public static FileAttribute<Set<PosixFilePermission>> getDefaultPosixFilePermiss
...
-     * @param stringSeparator the string separator
-     * @return the string
+     * @param fileName the file name
+     * @return the resources path
      */
-    public static String concatStringIfValueDontExist(String originalString, String stringToConcat, String stringSeparator) {
-        if (originalString.isEmpty() || originalString.isBlank()) {
-            stringSeparator = "";
+    @NonNull
+    public static Path getResourcesPath(String fileName) {
```
Comment (k-wall) [inline kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java:42]: Classpath resources might be within Jar files, so Path is an inappropriate choice of return type.  URL or URI would be better.

File: pom.xml [hunk 3]
```diff
@@ -61,7 +61,7 @@
         <jacoco-maven-plugin.version>0.8.12</jacoco-maven-plugin.version>
 
         <!-- Test dependencies -->
-        <strimzi.version>0.40.0</strimzi.version>
+        <strimzi.version>0.41.0</strimzi.version>
```
Comment (k-wall) [inline pom.xml:64]: You are upgrading Strimzi, the PR title should reflect this fact too.

File: kroxylicious-systemtests/src/main/resources/helm_strimzi_overrides.yaml [hunk 4]
```diff
@@ -0,0 +1,10 @@
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+# Helm Overrides File for Strimzi used by the system tests.
+replicas: 1
+featureGates: "+UseKRaft"
```
Comment (k-wall) [inline kroxylicious-systemtests/src/main/resources/helm_strimzi_overrides.yaml:9]: why do we need this line?  Isn't KRaft being enabled the default https://github.com/strimzi/strimzi-kafka-operator/blob/main/CHANGELOG.md#0400

File: kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java [hunk 5]
```diff
@@ -26,21 +33,24 @@ public static FileAttribute<Set<PosixFilePermission>> getDefaultPosixFilePermiss
...
+            if (resource == null) {
+                throw new IllegalArgumentException("Cannot find resource " + fileName + " on classpath");
+            }
+            overrideFile = resource.toURI();
         }
-
-        if (!originalString.contains(stringToConcat)) {
-            originalString = originalString.concat(stringSeparator + stringToConcat);
+        catch (URISyntaxException e) {
+            throw new IllegalStateException("Cannot determine file system path for " + resource);
```
Comment (k-wall) [inline kroxylicious-systemtests/src/main/java/io/kroxylicious/systemtests/utils/TestUtils.java:52]: chaining exceptions is almost always best practice.

```suggestion
            throw new IllegalStateException("Cannot determine file system path for " + resource, e);
```

**Comment by franvila** [comment]
@strimzi-ci run system tests

**Review by k-wall — APPROVED** [review]
lgtm.  tests passing on my machine.

**Pull request merged**